src/formatGeneratedModule.ts

```diff
@@ -23,13 +22,23 @@ export const formatterFactory = (
   if (compilerOptions.noImplicitAny) {
     nodeStatement = addAnyTypeCast(nodeStatement).trim();
   }
+  const devOnlyAssignmentsText =
+    devOnlyAssignments != null && devOnlyAssignments.length > 0
+      ? `\nif (process.env.NODE_ENV !== 'production') {\n  ${devOnlyAssignments.replace(
+          // `devOnlyAssignments` uses a flow comment, we can simply replace it with a
+          // ts cast.
+          "(node/*: any*/)",
+          "(node as any)"
+        )}\n}`
+      : "";
```

Comment:
I think we should just mimic the upstream behaviour, which is to always do this when queries are being persisted https://github.com/facebook/relay/blob/42033ebcc298af7e8ed90632b70077b7545a282d/packages/relay-compiler/codegen/writeRelayGeneratedFile.js#L107